crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -731,149 +740,274 @@ export.faucet_is_non_fungible_asset_issued
     # => [is_issued, pad(15)]
 end
 
-### NOTE ########################################
+### INPUT NOTE ##################################
 
-#! Returns the number of assets and the assets commitment of the note currently being processed.
+#! Returns the assets information of the specified input note.
 #!
-#! Inputs:  [pad(16)]
+#! Inputs:  [is_current_note, note_index, pad(14)]
 #! Outputs: [ASSETS_COMMITMENT, num_assets, pad(11)]
 #!
 #! Where:
-#! - num_assets is the number of assets in the note currently being processed.
-#! - ASSETS_COMMITMENT is a sequential hash of the assets in the note currently being processed.
+#! - is_current_note is the boolean flag indicating whether we should return the assets data from
+#!   the currently processing note or from the note with the specified index.
```

Comment:
Not for this PR, but I think we should switch from "currently processing note" to "active note". And then `is_current_note` would become `is_active_note` as well. But again, let's do this in a follow-up.